src/EnergyPlus/Autosizing/SystemAirFlowSizing.cc

```diff
@@ -751,8 +751,11 @@ Real64 SystemAirFlowSizer::size(EnergyPlusData &state, Real64 _originalValue, bo
                 if (this->dataHRFlowSizingFlag) { // HX sizing
                     if (this->curOASysNum) {
                         // size to supply air duct flow rate
-                        if (this->finalSysSizing(this->curSysNum).DesOutAirVolFlow > 0.0) {
+                        if (this->airLoopSysFlag && this->finalSysSizing(this->curSysNum).DesOutAirVolFlow > 0.0) {
                             this->autoSizedValue = this->finalSysSizing(this->curSysNum).DesOutAirVolFlow;
+                        } else if (this->outsideAirSys(this->curOASysNum).AirLoopDOASNum > -1) {
+                            auto &thisAirloopDOAS = this->airloopDOAS[this->outsideAirSys(this->curOASysNum).AirLoopDOASNum];
+                            this->autoSizedValue = thisAirloopDOAS.SizingMassFlow / state.dataEnvrn->StdRhoAir;
```

Comment:
The defect file has 5 AirloopHVAC objects and 1 AirloopHVAC:DedicatedOutdoorAirSystem which means FinalSysSizing is allocated to 5 and when CurSysNum = 6 (meaning the 6th outdoor air system associated with the DOAS) an array bounds error shows in the debugger. This method is used in other autosizing functions.